ENG-4085: docs for RFC 8707 Resource Indicators support (#4320)
FusionAuth/fusionauth-site: astro/src/content/docs/_shared/_application-oauth-settings.mdx

   <OAuthWildcardUsage fieldName="Authorized request origin URLs" wildcard="Allow wildcards" />
  </APIField>
  <APIField name="Authorized resource URIs" optional since="1.67.0">

>     An optional list of allowed resource server URIs for this Application, per RFC 8707 Resource Indicators for OAuth 2.0. Defaults to an empty list.

Reviewer: ```suggestion
    An optional list of allowed resource server URIs for this Application, per [RFC 8707 (Resource Indicators for OAuth 2.0)](https://www.rfc-editor.org/rfc/rfc8707.html). Defaults to an empty list.
```
Might be best to link here, since not everyone will have this memorized or handy :)

>     Each URI must be an absolute URI and may not contain a fragment (`#`). When this list is non-empty, clients may pass a `resource` parameter during the OAuth authorize and token flows to request access tokens scoped to a specific resource server. Only URIs present in this list are accepted; any other value returns an `invalid_target` error.

Reviewer: ```suggestion
    Each URI must be an absolute URI and may not contain a fragment (`#`). When this list is non-empty, clients may pass a `resource` parameter during the OAuth authorize and token flows to request access tokens scoped to a specific resource server. These flows only accept URIs present in this list; all other values return an `invalid_target` error.
```
suggestion: passive -> active voice

>     When this list is empty (the default), any `resource` parameter sent by the client is silently ignored and existing application behavior is unchanged.

Reviewer: ```suggestion
    When this list is empty (the default), FusionAuth silently ignores any `resource` parameter sent by the client; existing application behavior is unchanged.
```
suggestion passive -> active voice
